### PR TITLE
[GruntV0] Migrated legacy test 

### PR DESCRIPTION
**Pull request by max-zaytsev**
**Task name**: GruntV0

**Description**: Migrated legacy test

**Documentation changes required:** No

**Added unit tests:** Yes

**Attached related issue:** [#667](https://github.com/microsoft/build-task-team/issues/667)

**Related PR:** https://github.com/microsoft/azure-pipelines-tasks/pull/14204

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by EzzhevNikita — APPROVED** [review]

File: Tasks/GruntV0/Tests/L0.ts [hunk 1]
```diff
@@ -1,16 +1,244 @@
...
 import path = require('path');
+import * as ttm from 'azure-pipelines-task-lib/mock-test';
+
+const isWin = os.type().match(/^Win/);
 
 describe('GruntV0 Suite', function () {
-    before(() => {
+    this.timeout(parseInt(process.env.TASK_TEST_TIMEOUT) || 30000);
+
+    before((done) => {
```
Comment (DaniilShmelev) [inline Tasks/GruntV0/Tests/L0.ts:12]: would you mind adding type annotation here?

**Review by DaniilShmelev — APPROVED** [review]

File: Tasks/GruntV0/Tests/L0.ts (hunk 1 above)
Comment (max-zaytsev) [reply Tasks/GruntV0/Tests/L0.ts]: Yes, sure. Thanks!

File: Tasks/GruntV0/Tests/L0GruntCliNotSet.ts [hunk 2]
```diff
@@ -1,30 +1,39 @@
...
         },
         "/usr/local/bin/node ./node_modules/istanbul/lib/cli.js cover --report cobertura --report html -i ./**/build/src/*.js ./node_modules/mocha/bin/_mocha **/build/test/*.js": {
-            "code": 1,
+            "code": 0,
             "stdout": "istanbul output here",
-            "stderr": "istanbul failed with this output"
         },
         "/usr/local/bin/node ./node_modules/istanbul/lib/cli.js cover --report cobertura --report html -i .\\**\\build\\src\\*.js ./node_modules/mocha/bin/_mocha **/build/test/*.js": {
-            "code": 1,
+            "code": 0,
```
Comment (anatolybolshakov) [inline Tasks/GruntV0/Tests/L0GruntCliNotSet.ts:34]: Is it expected that it's succeded currently, but was failed before?

File: Tasks/GruntV0/task.loc.json [hunk 3]
```diff
@@ -16,8 +16,8 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 0,
-    "Minor": 166,
-    "Patch": 2
+    "Minor": 182,
```
Comment (anatolybolshakov) [inline Tasks/GruntV0/task.loc.json:19]: Are these redundant changes related to node 10 migration?

File: Tasks/GruntV0/Tests/L0GruntCliNotSet.ts (hunk 2 above)
Comment (max-zaytsev) [reply Tasks/GruntV0/Tests/L0GruntCliNotSet.ts:34]: There was no `L0GruntCliNotSet.ts` file before. It's just how github shows diffs
image

File: Tasks/GruntV0/task.loc.json (hunk 3 above)
Comment (max-zaytsev) [reply Tasks/GruntV0/task.loc.json:19]: This branch is based on node10 migration branch. I was going to merge it first
